File: src/main/java/com/homecraft/api/entity/Order.java
package com.homecraft.api.entity;

import jakarta.persistence.*;
import java.math.BigDecimal;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "orders")
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "product_id", nullable = false)
    private Integer productId;

    @Column(name = "customer_id", nullable = false)
    private Integer customerId;

    @Column(name = "seller_id", nullable = false)
    private Integer sellerId;

    @Column(nullable = false, length = 100)
    private String name;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal price;

    @Column(nullable = false)
    private Integer quantity;

    @Column(length = 255)
    private String customizations;

    @Column(nullable = false, length = 20)
    private String status;

}

File: HomeCraft.API/Models/Order.cs
public class Order
{
    public int Id { get; set; }
    public int ProductId { get; set; }
    public int CustomerId { get; set; }
    public int SellerId { get; set; }

    public string Name { get; set; }
    public decimal Price { get; set; }
    public int Quantity { get; set; }
    public string Customizations { get; set; }
    public string Status { get; set; }
}
